### PR TITLE
Added an initial logging util that always logs agent components

### PR DESCRIPTION
**Pull request by snova-codym**
Initial logging utility.  We may wish to make this an optional method next.

### CODE REVIEW

File: agent_workflows/data/uber_lyft.chromadb/bb7f200d-6c4e-4ae2-9441-2057e145c3fd/data_level0.bin
Comment (snova-petrojm) [inline agent_workflows/data/uber_lyft.chromadb/bb7f200d-6c4e-4ae2-9441-2057e145c3fd/data_level0.bin:1]: Do we need to push the uber_lyft.chromadb/bb7f200d-6c4e-4ae2-9441-2057e145c3fd/ ?

**Comment by snova-codym** [comment]
Can remove it - I just thought it would be helpful for reproducibility, particularly for the notebooks.  Understand that larger files like this aren't necessarily good practice, though.  They are small, but happy to remove.

**Comment by snova-kwasia** [comment]
@snova-codym - Have you resolved @snova-petrojm comment here, just checking whether we're good to merge.

**Comment by snova-codym** [comment]
No because I don't know how we want to proceed.  Just need a yay or nay on removing it.  :).  I'm fine either way.  After that, I believe @snova-francescar will do some review as well before merging.

**Comment by snova-petrojm** [comment]
@snova-kwasia , I dont think we pushed vector DBs previously to main, right? If that's the case, my recommendation would be to remove it. I think the kit already provides sample PDF files (data/lyft.pdf and data/uber.pdf) that users can use. 
Note I haven't reviewed the rest of the PR, I think @snova-francescar will do so. 
Thanks.

**Comment by snova-kwasia** [comment]
Ok thank you @snova-codym - Let's remove the vectordb please, you're right @snova-petrojm we don't push these to main.

File: utils/rag/rag_components.py [hunk 1]
```diff
@@ -720,7 +738,7 @@ def determine_cont(self, state: dict) -> str:
         """
 
         subquestions: List[str] = state['subquestions']
-        print(len(subquestions))
+        # print(len(subquestions))
```
Comment (snova-francescar) [inline utils/rag/rag_components.py:741]: Should we remove this line?

**Review by snova-francescar — APPROVED** [review]
Great solution for logging!

**Comment by snova-codym** [comment]
I have made my minor updates.  @snova-jamesv could you do a final pass review and merge?  You will see a conflict for base_components.  Accept what is in the repo currently, please.

**Pull request merged**